src/main/java/org/opensearch/ad/rest/RestAnomalyDetectorJobAction.java

```diff
@@ -99,11 +100,6 @@ protected ExtensionRestResponse prepareRequest(RestRequest request) throws IOExc
             throw new IllegalStateException(CommonErrorMessages.DISABLED_ERR_MSG);
         }
 
-        // Ensure job details are registered with Job Scheduler prior to creating a job
-        if (!registeredJobDetails) {
```

Comment:
It doesn't look like the previous version conditioned on it being a start job or having content. For my own education, why is it necessary for us to add that in the conditional (vs. handling these conditions in the`registerJobDetails()` method itself?)

Reply:
I can add these checks in the `registerJobDetails` method. Previously it was not necessary to check if the request was a start job request or if the request had content. Now that we've enabled historical analysis, it is necessary to prevent these routes from registering job details with Job Scheduler as only real time analysis needs to trigger this registration.